Source/SourceKittenFramework/UID.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  UID.swift
+//  SourceKitten
+//
+//  Created by Norio Nomura on 2/07/18.
+//  Copyright © 2018 SourceKitten. All rights reserved.
+//
+
+import Foundation
+#if SWIFT_PACKAGE
+    import SourceKit
+#endif
+
+/// Swift representation of sourcekitd_uid_t
+public struct UID {
+    let uid: sourcekitd_uid_t
+    init(_ uid: sourcekitd_uid_t) {
+        self.uid = uid
+    }
+
+    public init(_ string: String) {
+        self.init(sourcekitd_uid_get_from_cstr(string)!)
+    }
+
+    public init<T>(_ rawRepresentable: T) where T: RawRepresentable, T.RawValue == String {
+        self.init(rawRepresentable.rawValue)
+    }
+
+    var string: String {
+        return String(cString: sourcekitd_uid_get_string_ptr(uid)!)
+    }
+}
+
+extension UID: CustomStringConvertible {
+    public var description: String {
+        return string
+    }
+}
+
+extension UID: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self.init(value)
+    }
+}
+
+extension UID: Hashable {
```

Comment:
Thoughts on using the auto-synthesized `Hashable` implementation for Swift 4.1+?

```swift
extension UID: Hashable {
#if !swift(>=4.1)
    public var hashValue: Int {
        return uid.hashValue
    }

    public static func == (lhs: UID, rhs: UID) -> Bool {
        return lhs.uid == rhs.uid
    }
#endif
}
```

I don't feel strongly, but it seems like it might be easier to find code that can be removed once we require Swift 4.1+ to compile SourceKitten.